### PR TITLE
dynamic setting of stats_dump_period_sec through SetDBOption()

### PR DESCRIPTION
**Pull request by raza15**
Resolved the following issue: https://github.com/facebook/rocksdb/issues/1930

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -44,6 +44,7 @@
 * Support dynamically change `delayed_write_rate` and `max_total_wal_size` options via SetDBOptions().
 * Introduce DB::DeleteRange for optimized deletion of large ranges of contiguous keys.
 * Support dynamically change `delayed_write_rate` option via SetDBOptions().
+* Support dynamically change `stats_dump_period_sec` option via SetDBOptions().
```
Comment (yiwu-arbug) [inline HISTORY.md]: Please add the changelog to Unreleased section, which will be included in next release.

File: util/db_options.h [hunk 2]
```diff
@@ -94,6 +93,7 @@ struct MutableDBOptions {
   uint64_t delayed_write_rate;
   uint64_t max_total_wal_size;
   uint64_t delete_obsolete_files_period_micros;
+  unsigned int stats_dump_period_sec;
```
Comment (yiwu-arbug) [inline util/db_options.h]: Remove the same field from ImmutableDBOptions.

File: db/db_impl.cc [hunk 3]
```diff
@@ -618,12 +618,12 @@ static void DumpMallocStats(std::string* stats) {
 #endif  // !ROCKSDB_LITE
 
 void DBImpl::MaybeDumpStats() {
-  if (immutable_db_options_.stats_dump_period_sec == 0) return;
+  if (mutable_db_options_.stats_dump_period_sec == 0) return;
```
Comment (yiwu-arbug) [inline db/db_impl.cc]: mutable_db_options_ is guarded by DBImpl::mutex_ but MaybeDumpStats() is called without holding mutex. So there will be race condition on the option if another thread call SetDBOptions. You need to lock the mutex, get the value of stats_dump_period_sec, and unlock mutex to finish the rest of work here.

File: HISTORY.md (hunk 1 above)
Comment (raza15) [inline HISTORY.md]: Done

File: util/db_options.h (hunk 2 above)
Comment (raza15) [inline util/db_options.h]: This is already done.

File: db/db_impl.cc (hunk 3 above)
Comment (raza15) [inline db/db_impl.cc]: Done.

**Comment by raza15** [comment]
@yiwu-arbug please see the updated PR

File: HISTORY.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # Rocksdb Change Log
 ## Unreleased
+* Support dynamically change `stats_dump_period_sec` option via SetDBOptions().
```
Comment (yiwu-arbug) [inline HISTORY.md]: Add "## Public API Change"

File: util/db_options.h (hunk 2 above)
Comment (yiwu-arbug) [inline util/db_options.h]: My bad. I don't know why I didn't see the change.

**Comment by yiwu-arbug** [comment]
Tests looking good. The only travis failure looks unrelated. @raza15 can you run `make format` and update the PR? Thanks!

**Comment by raza15** [comment]
@yiwu-arbug I am getting the floowing error when I run ``` make format ``` :
```
You didn't have clang-format-diff.py available in your computer!
You can download it by running: 
    curl --location http://goo.gl/iUW1u2 -o clang-format-diff.py
and move clang-format-diff.py to some directory within PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
```
I have downloaded clang-format-diff.py and put it in my path directory. But I still get the error. Do you know how to solve this?

**Comment by yiwu-arbug** [comment]
No idea :( You can apply this command directly:
```git diff -U0 $LAST_MASTER^ | $CLANG_FORMAT_DIFF -i -p 1```

**Comment by raza15** [comment]
@yiwu-arbug updated the PR after running ``` make format ```. I'll start a new PR to update the instructions for running clang.

**Comment by yiwu-arbug** [comment]
Merging. Thank you again for the contribution!

**Pull request closed without merging**